packages/flow-core/src/components/f-emoji-picker/f-emoji-picker.ts

```diff
@@ -0,0 +1,490 @@
+	 */
+	get excludeCategories() {
+		if (this.exclude && this.exclude.length > 0) {
+			const excludeSet = new Set(this.exclude);
+			const displayCategories = this.categories.filter(category => {
+				return !excludeSet.has(category as FEmojiPickerCategories);
+			});
+			return displayCategories;
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * categories to be displayed for picker according to inclide amd exclude
+	 */
+	get categroiesToDisplay() {
+		return this.include && this.include.length > 0
+			? this.include
+			: this.excludeCategories.length > 0
+			? this.excludeCategories
+			: this.categories;
+	}
+
+	/**
+	 * header slot display
+	 */
+	headerSectionDisplay() {
+		if (!this.hasLabel && !this.hasDescription) {
+			this.emojiPickerHeader.style.display = "none";
+		} else {
+			this.emojiPickerHeader.style.display = "";
+		}
+		if (!this.hasLabel) {
+			this.labelSlot.style.display = "none";
+		}
+	}
+
+	/**
+	 * help section display
+	 */
+	helpSectionDisplay() {
+		if (!this.hasHelperText) {
+			this.emojiPickerError.style.display = "none";
+		} else {
+			this.emojiPickerError.style.display = "";
+		}
+	}
+
+	/**
+	 * open/close picker
+	 * @param value boolean
+	 */
+	toggleEmojiPicker(value: boolean) {
+		this.emojiPickerPopover.target = this.emojiPicker;
+		this.emojiPickerPopover.open = value;
+	}
+
+	/**
+	 *
+	 * @param valuePicked emoji as value
+	 */
+	handleSelectEmoji(valuePicked?: Emoji) {
+		// e.stopPropagation();
+		/**
+		 * @event input
+		 */
+		const event = new CustomEvent("input", {
+			detail: {
+				value: valuePicked ? valuePicked.native ?? valuePicked.src : ""
+			},
+			bubbles: true,
+			composed: true
+		});
+		this.value = valuePicked ? valuePicked.native ?? valuePicked.src : "";
+		this.dispatchEvent(event);
+		if (this["close-on-select"]) {
+			this.toggleEmojiPicker(false);
+		}
+	}
+
+	/**
+	 * clear emoji value
+	 * @param e MouseEvent
+	 */
+	clearValue(e: MouseEvent) {
+		e.stopPropagation();
+		this.handleSelectEmoji();
+	}
+
+	/**
+	 * add recently used emojis
+	 */
+	handleFrequentEmojiCategory() {
+		if (this.recent && this.recent?.length > 0) {
+			(data as EmojiMartData).categories.forEach((item: Category) => {
+				if (item.id === "frequent") {
+					item.emojis = this.recent as string[];
+				}
+			});
+		}
+	}
+
+	/**
+	 * handle custom emoji insertion
+	 */
+	handleCustomEmoji() {
+		if (this.custom && this.custom?.length > 0) {
+			this.custom?.forEach(item => {
+				(data as EmojiMartData).categories.push(item);
+			});
+		}
+	}
+
+	/**
+	 * validation for placeolder
+	 */
+	validateProperties() {
+		const emojiRegex = /\p{Extended_Pictographic}/u;
+		if (this.placeholder) {
+			if (!emojiRegex.test(this.placeholder)) {
+				throw new Error("f-emoji-picker : placeholder should be an emoji");
+			}
+		}
+	}
+
+	render() {
+		this.validateProperties();
+
+		/**
+		 * click outside the f-emoji wrapper area
+		 */
+		window.addEventListener("click", (e: MouseEvent) => {
```

Comment:
You need named function to do addEventListner.

And removeEventListnee in disconnectedCallback